pytravharv/SubjPropPathAssertion.py

```diff
@@ -120,13 +110,21 @@ def _assert_at_depth(self):
             "Asserting a property path for a given subject at a given depth"
         )
         log.debug("Depth: {}".format(self.max_depth - self.current_depth))
-        SPARQLQuery = self._sparql_trajectory_check(
-            self.max_depth - self.current_depth
-        )
-        if self.target_store.verify(SPARQLQuery):
+        if self.rdf_store_access.verify(
+            self.subject,
+            self.assertion_path.get_path_for_depth(
+                self.max_depth - self.current_depth
+            ),
+            self.prefix_set,
+        ):
             self._harvest_and_surface()
             return
-        self.target_store.ingest(WebAccess(self.subject), self.config_filename)
+        print(f"adding {self.subject} to {self.graph_name}")
+        graph_to_ingest = web_access(self.subject)
```

Comment:
suggested better name for `web_access` might be something like `find_description_graph(subject)` 

and let the documentation explain how various strategies are being applied (i.e. conneg, fair-signposting, ...) 

also feels like this function has great potential to get some wider usage, that might require plugable / configurable strategies in the future? (but then we can introduce that flexibility later)